test/milton/MiltonEvents.t.sol

```diff
@@ -0,0 +1,526 @@
+                iporPublicationFee: TestConstants.TC_IPOR_PUBLICATION_AMOUNT_18DEC, // iporPublicationFee
+                liquidationDepositAmount: TestConstants.TC_LIQUIDATION_DEPOSIT_AMOUNT_18DEC // liquidationDepositAmount
+            }), // money
+            block.timestamp, // openTimestamp
+            block.timestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, // endTimestamp, 28 days, PERIOD_28_DAYS_IN_SECONDS
+            MiltonTypes.IporSwapIndicator({
+                iporIndexValue: TestConstants.PERCENTAGE_3_18DEC, // iporIndexValue
+                ibtPrice: 1 * TestConstants.D18, // ibtPrice
+                ibtQuantity: 99670098970308907327800, // ibtQuantity
+                fixedInterestRate: TestConstants.PERCENTAGE_4_18DEC // fixedInterestRate, 4%
+            }) // indicator
+        );
+        miltonDai.itfOpenSwapPayFixed(
+            block.timestamp, // openTimestamp
+            TestConstants.USD_10_000_18DEC, // totalAmount
+            TestConstants.PERCENTAGE_6_18DEC, // acceptableFixedInterestRate, 6%
+            TestConstants.TC_IPOR_PUBLICATION_AMOUNT_18DEC // leverage, LEVERAGE_18DEC
+        );
+    }
+
+    function testShouldEmitEventWhenOpenReceiveFixedSwap18Decimals() public {
+        // given
+        MockTestnetToken daiMockedToken = getTokenDai();
+        ItfIporOracle iporOracle = getIporOracleAsset(_userOne, address(daiMockedToken), 0);
+        IpToken ipTokenDai = getIpTokenDai(address(daiMockedToken));
+        MockCase0Stanley stanleyDai = getMockCase0Stanley(address(daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        ItfMiltonDai miltonDai = getItfMiltonDai(
+            address(daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        ItfJosephDai josephDai = getItfJosephDai(
+            address(daiMockedToken),
+            address(ipTokenDai),
+            address(miltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, daiMockedToken, address(josephDai), address(miltonDai));
+        prepareMilton(miltonDai, address(josephDai), address(stanleyDai));
+        prepareJoseph(josephDai);
+        prepareIpToken(ipTokenDai, address(josephDai));
+        // when
+        _miltonSpreadModel.setCalculateQuoteReceiveFixed(TestConstants.PERCENTAGE_2_18DEC); // 2%
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(daiMockedToken), TestConstants.PERCENTAGE_3_18DEC, block.timestamp); // 3%, PERCENTAGE_3_18DEC
+        vm.prank(_liquidityProvider);
+        josephDai.itfProvideLiquidity(TestConstants.USD_28_000_18DEC, block.timestamp); // TestConstants.USD_28_000_18DEC
+        vm.prank(_userTwo);
+        vm.expectEmit(true, true, false, false);
```

Comment:
Why do you use vm.expectEmit(true, true, false, false) and not vm.expectEmit(true, true, true, true)